### PR TITLE
Add speechmatics

### PR DESCRIPTION
**Pull request by TudorCRL**

### CODE REVIEW

File: examples/audio_transcribe.py [hunk 1]
```diff
@@ -13,6 +13,16 @@
 with sr.AudioFile(AUDIO_FILE) as source:
     audio = r.record(source)  # read the entire audio file
 
+# recognize speech using Speechmatics
+SPEECHMATICS_KEY = "INSERT SPEECHMATICS API KEY HERE"
+try:
+    print("Speechmatics thinks you said " + r.recognize_ibm(audio, key=SPEECHMATICS_KEY))
```
Comment (jamied157) [inline examples/audio_transcribe.py]: `recognize_ibm` -> `recognize_speechmatics`

File: examples/extended_results.py [hunk 2]
```diff
@@ -16,6 +16,16 @@
 with sr.AudioFile(AUDIO_FILE) as source:
     audio = r.record(source)  # read the entire audio file
 
+# recognize speech using Speechmatics
+SPEECHMATICS_KEY = "INSERT SPEECHMATICS API KEY HERE"
+try:
+    print("Speechmatics results:")
+    pprint(r.recognize_sphinx(audio, key=SPEECHMATICS_KEY, transcript_format="json-v2"))
```
Comment (jamied157) [inline examples/extended_results.py]: `sphinx` -> `speechmatics`

File: speech_recognition/__init__.py [hunk 3]
```diff
@@ -1702,6 +1702,47 @@ def recognize_vosk(self, audio_data, language='en'):
...
+
+        try:
+            from speechmatics.models import ConnectionSettings, BatchTranscriptionConfig
+            from speechmatics.batch_client import BatchClient
+            from speechmatics.constants import BATCH_SELF_SERVICE_URL
+        except:
+            raise RequestError("missing speechmatics python module: install using `pip install speechmatics-python`")
+
+        wav_data = audio_data.get_wav_data(
+            convert_rate=None if audio_data.sample_rate >= 16000 else 16000  # audio samples must be at least 16 kHz
```
Comment (jamied157) [inline speech_recognition/__init__.py]: You might know better than me - is this correct? Do we deal with sample rate >= 16000 ourselves?

**Review by jamied157 — CHANGES_REQUESTED** [review]
Looks good, few small things in the examples and I don't know how we deal with sample rate

File: examples/extended_results.py (hunk 2 above)
Comment (TudorCRL) [reply examples/extended_results.py]: Hmmm bit sus, but sure

File: speech_recognition/__init__.py (hunk 3 above)
Comment (TudorCRL) [reply speech_recognition/__init__.py]: Not sure, let me check with some of the heads

File: examples/audio_transcribe.py (hunk 1 above)
Comment (TudorCRL) [reply examples/audio_transcribe.py]: I'm actual a double agent implanted by IBM and this was intentional. But now you've found me out so I'll change it

File: speech_recognition/__init__.py (hunk 3 above)
Comment (TudorCRL) [reply speech_recognition/__init__.py]: I asked Dan, he reckons we don't get anything less than 8000, but may be able to support lower than that. It's not documented anywhere that I can see. Maybe we just let the runtime deal with it and pass any errors to users?

Comment (jamied157) [reply speech_recognition/__init__.py]: Yeah sounds good to me, we can keep an eye on the repo and see if anyone is having issues if it gets merged

**Review by jamied157 — APPROVED** [review]

File: speech_recognition/__init__.py (hunk 3 above)
Comment (TudorCRL) [reply speech_recognition/__init__.py]: Removed that now, will resolve and merge and then open a PR into the upstream

**Pull request merged**